src/DsaBottomUp.cc

```diff
@@ -238,6 +310,12 @@ bool BottomUpAnalysis::runOnModule(Module &M, GraphMap &graphs) {
   size_t functionsProcessed = 0;
   size_t percentageProcessed = 0;
 
+  /// -- keep track of indirect call sites that have been copied to a
+  /// -- caller. This helps us to decide whether or not an indirect
+  /// -- callsite can be resolved: given function F, if a callsite C
+  /// -- is not copied to F's callers and F' address is never taken,
+  /// -- then the callsite C can be resolved.
+  std::set<DsaCallSite *> copiedCallSites;
```

Comment:
seems overly complex (although I don't see immediate flaw in the argument).

An indirect call site `call` is resolved if `cell(call)` is not reachable from globals and arguments.
This also means that `cell(call)` is not mapped by the cloner (which is the cache we can use).